### PR TITLE
Clean up metrics artf detail pages

### PR DESCRIPTION
**Pull request by likawind**
## Describe your changes and why you are making these changes
This PR cleans up artifact and metrics page. By managing large components individually, we reduced `index` file of each page from ~300+ lines to ~150 lines. Some components are also reusable. For example, `artifact/summaryList` can be reused for inputs / outputs for operator details page as well.

## Related issue number (if any)
ENG-1747

## Demo (if any)
Verified these refactored pages works properly.
Screen Shot 2022-09-26 at 11 51 49 PM
Screen Shot 2022-09-26 at 11 52 37 PM

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/pages/metric/id/index.tsx [hunk 1]
```diff
@@ -125,134 +117,18 @@ const MetricDetailsPage: React.FC<MetricDetailsPageProps> = ({
...
-              marker: { color: 'red' },
-            },
-          ]}
-          layout={{ width: '100%', height: '100%' }}
-        />
-        <PaginatedTable data={historicalData} />
-      </Box>
-    );
-  }
+  const inputs = operator.inputs
```
Comment (eunice-chan) [inline src/ui/common/src/components/pages/metric/id/index.tsx:120]: We can make a helper function for the inputs and outputs variables.
```
const filterArtifacts = (artifacts) => {
artifacts
    .map(
      (artifactId) =>
        (workflowDagResultWithLoadingStatus.result?.artifacts ?? {})[artifactId]
    )
    .filter((artf) => !!artf);

const inputs = filterArtifacts(operator.inputs);
const outputs = filterArtifacts(operator.outputs);
};

**Review by eunice-chan — APPROVED** [review]
Thanks for component-ifying everything! Looks very neat now

File: src/ui/common/src/components/workflows/artifact/content.tsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+}) => {
+  if (!artifact.result) {
+    return (
+      <Typography variant="h5" component="div" marginBottom="8px">
+        No result to show for this artifact.
+      </Typography>
+    );
+  }
+
+  if (artifact.result.content_serialized != null) {
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/content.tsx:30]: !==

File: src/ui/common/src/components/workflows/artifact/content.tsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        <Alert title="Cannot parse table data.">
+          {err}
+          {contentWithLoadingStatus.data}
+        </Alert>
+      );
+    }
+  }
+  // TODO: handle images here
+  return (
+    <Typography variant="body1" component="div" marginBottom="8px">
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/content.tsx:104]: Is there a reason why we're using component="div" in all the typography components here? Haven't seen that being used in too many other places.

File: src/ui/common/src/components/workflows/artifact/metric/history.tsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+      <Alert title="Failed to load historical data.">
+        {historyWithLoadingStatus.status.err}
+      </Alert>
+    );
+  }
+
+  const historicalData: Data = {
+    schema: {
+      fields: [
+        { name: 'status', type: 'varchar' },
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/metric/history.tsx:35]: maybe we should make this fields array into a constant.

Comment (agiron123) [reply src/ui/common/src/components/workflows/artifact/metric/history.tsx:35]: something like const metricsHistoricalSchema could be a good name.

File: src/ui/common/src/components/workflows/artifact/metric/history.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+  return (
+    <Box display="flex" justifyContent="center" flexDirection="column">
+      <Plot
+        data={[
+          {
+            x: timestamps,
+            y: values,
+            type: 'scatter',
+            mode: 'lines+markers',
+            marker: { color: 'red' },
```
Comment (agiron123) [inline src/ui/common/src/components/workflows/artifact/metric/history.tsx:66]: See if we can use the same dark blue color that we're using throughout our app here. Might be a nice touch 🧑‍🎨

**Review by agiron123 — APPROVED** [review]
Nice work! Excited to see these changes go in.

File: src/ui/common/src/components/workflows/artifact/content.tsx (hunk 2 above)
Comment (likawind) [reply src/ui/common/src/components/workflows/artifact/content.tsx:30]: let me add a comment, this is intentional as it makes sure the field is neither `null` nor `undefined`

File: src/ui/common/src/components/workflows/artifact/content.tsx (hunk 3 above)
Comment (likawind) [reply src/ui/common/src/components/workflows/artifact/content.tsx:104]: I don't know this is copied from your original version of artf detail page ;)

File: src/ui/common/src/components/workflows/artifact/metric/history.tsx (hunk 5 above)
Comment (likawind) [reply src/ui/common/src/components/workflows/artifact/metric/history.tsx:66]: which string should I put here?

**Comment by vsreekanti** [comment]
This may be out of scope for this PR, but we should do a quick design review on these new pages. There's some rough edges that can be ironed out. e.g., there shouldn't be colons in headings, and it's not clear why these two cards are different heights or have dividers underneath them:
 
image

Similarly the height of the data table should be adaptive so when there's less data, there shouldn't be dead space at the bottom: 
image

**Comment by likawind** [comment]
@vsreekanti yes, these pages are expected in more rough state for now. Once all pages are ready (which should happen soon) we will ask feedbacks from the team and polish these places altogether

**Comment by agiron123** [comment]
@likawind there's a number that you can set here for rows per page. If it's less than 10, we should use that number. Currently the minimum number of entries per page is set to 5, but that's something that we can adjust.

**Comment by likawind** [comment]
Ahh figured. Though I do agree with Vikram that the size should be adaptable if there are only 5 or 10 rows (now I also need to scroll for just 10 rows which may not be ideal as well).

Let's revisit this as a separate problem, I think we need to play with the component a bit more

**Comment by likawind** [comment]
https://linear.app/aqueducthq/issue/ENG-1752/make-data-tables-size-more-adaptive

**Comment by likawind** [comment]
https://linear.app/aqueducthq/issue/ENG-1751/refinements-for-input-output-list

File: src/ui/common/src/components/workflows/artifact/metric/history.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  return (
+    <Box display="flex" justifyContent="center" flexDirection="column">
+      <Plot
+        data={[
+          {
+            x: timestamps,
+            y: values,
+            type: 'scatter',
+            mode: 'lines+markers',
+            // colors are 'blue.900'. Plot doesn't seem to accept color from theme.
```
Comment (likawind) [inline src/ui/common/src/components/workflows/artifact/metric/history.tsx:68]: @agiron123 I don't think plotly like `blue.900` so I put the hex here. lmk if there are better ways to solve this

Comment (agiron123) [reply src/ui/common/src/components/workflows/artifact/metric/history.tsx:68]: Yeah, import the theme object and use template literals. This is how we normally deal with react components that do not have the sx prop or our theme context.

```
color: `${theme.blue900}`
```
or however it should be indexed.

**Pull request merged**